app/src/main/java/com/tangem/tap/domain/userWalletList/implementation/GeneralUserWalletsListManager.kt

```diff
@@ -0,0 +1,155 @@
+package com.tangem.tap.domain.userWalletList.implementation
+
+import com.tangem.common.CompletionResult
+import com.tangem.datasource.local.preferences.AppPreferencesStore
+import com.tangem.datasource.local.preferences.PreferencesKeys
+import com.tangem.datasource.local.preferences.utils.get
+import com.tangem.domain.wallets.legacy.UserWalletsListManager
+import com.tangem.domain.wallets.models.UserWallet
+import com.tangem.domain.wallets.models.UserWalletId
+import com.tangem.utils.coroutines.CoroutineDispatcherProvider
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.*
+import timber.log.Timber
+
+/**
+ * General implementation of [UserWalletsListManager] that helps to switch between Runtime and Biometric
+ * implementations.
+ *
+ * @property runtimeUserWalletsListManager   runtime user wallets list manager
+ * @property biometricUserWalletsListManager biometric user wallets list manager
+ * @property appPreferencesStore             app preferences store
+ * @property dispatchers                     coroutine dispatcher provider
+ *
+ * @author Andrew Khokhlov on 09/02/2024
+ */
+internal class GeneralUserWalletsListManager(
+    private val runtimeUserWalletsListManager: UserWalletsListManager,
+    private val biometricUserWalletsListManager: UserWalletsListManager,
+    private val appPreferencesStore: AppPreferencesStore,
+    private val dispatchers: CoroutineDispatcherProvider,
+) : UserWalletsListManager.Lockable {
+
+    private val applicationScope = CoroutineScope(dispatchers.io)
+    private val implementation = MutableStateFlow(runtimeUserWalletsListManager)
+
+    init {
+        subscribeOnCurrentManager()
+    }
+
+    override val userWallets: Flow<List<UserWallet>>
+        get() = channelFlow {
+            implementation.collectLatest {
+                it.userWallets.collectLatest(::send)
+            }
+        }
```

Comment:
Вроде тут, и ниже, можно упростить, через `flatMapLatest`:

```suggestion
        get() = implementation
            .flatMapLatest { it.userWallets }
```

Reply:
Отличная замечание, спасибо